comptests/s3/test_user_policy_crud.py

```diff
@@ -0,0 +1,424 @@
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+# pylint: disable=E0401
+"""All IAM users test  Module."""
+import json
+import asyncio
+import time
+import logging
+from http import HTTPStatus
+import pytest
+
+from libs.s3.s3_iam_rest_rgw import RestApiRgw
+from commons import cortxlogging
+
+class TestRgwPolicyCrud:
+    """
+    REST API Test cases for IAM users.
+    IAM CRUD operations.
+    """
+    @classmethod
+    def setup_class(cls):
+        """Function will be invoked before running each test case."""
+        cls.log = logging.getLogger(__name__)
+        cls.log.info("STARTED: setup class ")
+        cls.obj = RestApiRgw()
+        cls.user_name_prefix = "user"
+        cls.access_key_prefix = "access_key"
+        cls.secret_key_prefix = "secret_key"
+        cls.created_users = []
+        cls.tenant = 'Group1'
+
+    def teardown_method(self):
+        """
+        Teardown for deleting resources like users,object and bucket created as part of testcases
+        """
+        self.log.info("[STARTED] ######### Teardown #########")
+        self.log.info("Deleting all users created as part of test")
+        delete_failed = []
+        delete_success = []
+        self.log.debug("created_users list : %s",self.created_users)
+        for usr in self.created_users:
+            self.log.info("Sending request to delete user %s", usr)
+            loop = asyncio.get_event_loop()
+            status = loop.run_until_complete(self.obj.delete_user(usr))
+            if status != HTTPStatus.OK:
+                delete_failed.append(usr)
+            else:
+                delete_success.append(usr)
+        for usr in delete_success:
+            self.created_users.remove(usr)
+        self.log.info("User delete success list %s", delete_success)
+        self.log.info("User delete failed list %s", delete_failed)
+
+    def create_user(self,user_params):
+        """ Create user function. """
+        loop = asyncio.get_event_loop()
+        status, user_info = loop.run_until_complete(self.obj.create_user(user_params))
+        return status, user_info
+
+    def put_policy(self,user_params):
+        """ Apply Policy to the user function. """
+        loop = asyncio.get_event_loop()
+        status = loop.run_until_complete(self.obj.put_user_policy(user_params))
```

Comment:
You could have removed asyncio event loop.  run_until_complete is not going to retry in case of failure or socked exception.